content/en/_index.md

```diff
@@ -218,6 +301,19 @@ a file or a specific action is not.
         // Snippet content ...
     }
   {{< /tab >}}
+  {{< tab header="Node.js" >}}
+    function main(requiredArg) {
+      // This argument is required, and should be added to main()
+      // const requiredArg = '...'
```

Comment:
`exampleSnippet` should have this arg, "main" should not.

Reply:
How do the tests pass in arguments to the snippet then?

(Would we have to add it to `exports`?)

Reply:
So I think if you are using this pattern:

```js

function main(){
      // ...
      exampleSample(arg1, arg2)
}

function exampleSample(arg1, arg2){
      // ...
}

if module.require == "main" {
     main()
}
```

Then the following should be true:
1. Main should only be a stub used by the user to set variables and run in whichever way they are most comfortable
2. testing can still happen on exampleSample directly, since the test can set args needed for it directly.

Reply:
I presume the _tests_ can `require(...)` a snippet file to invoke `main`?

(If so - how would the tests pass values to the sample itself? Or are the not supposed to?)

Reply:
I'm not quite sure what you're asking, but here's an example of this pattern in java-land (with one slight derivation that this before we used "main" - everything else is the same):

Here's the "main" method: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/c5df4758e5fbbb2261338e69fdfd1f8794a5f7d1/kms/src/main/java/kms/CreateKeyRing.java#L27

Here's the "exampleSnippet": https://github.com/GoogleCloudPlatform/java-docs-samples/blob/c5df4758e5fbbb2261338e69fdfd1f8794a5f7d1/kms/src/main/java/kms/CreateKeyRing.java#L36

Here's the test calling "exampleSnippet": https://github.com/GoogleCloudPlatform/java-docs-samples/blob/c5df4758e5fbbb2261338e69fdfd1f8794a5f7d1/kms/src/test/java/kms/SnippetsIT.java#L317

Reply:
There *is* parameter-passing going on there. In Java, that's easy - it's a `public` class with a `public` method.

In Node, to make a method `public` you have to add it to `exports` - otherwise it's (effectively) `private`.

Reply:
My only (minor) objection is wordiness - we'd have to do one of the following. Of the two, **Option B** feels cleaner to me.

**Option A**
```js
exports.exampleSnippet = (arg1, arg2) => {
   ...
}

function main() {
  arg1 = ...
  arg2 = ...

  // Notice the extra `exports` here
  exports.exampleSnippet(arg1, arg2);
}
```

**Option B**
```js
const exampleSnippet = (arg1, arg2) => {
  ...
}

function main() {
  arg1 = ...
  arg2 = ...

  // No `exports`! :)
  exampleSnippet(arg1, arg2)
}

exports.exampleSnippet = exampleSnippet;
```